.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,347 @@
+two together.
+
+Write your commit message in the present tense: "Fix bug" and not "Fixed
+bug."  This convention matches up with commit messages generated by
+commands like git merge and git revert.
+
+Further paragraphs come after blank lines.
+
+- Bullet points are okay, too
+- Typically a hyphen or asterisk is used for the bullet, preceded by a
+  single space, with blank lines in between, but conventions vary here
+- Use a hanging indent
+```
+
+Prefix the summary with the subsystem/package when possible. Many other
+projects make use of the code and this makes it easier for them to tell when
+something they're using has changed. Have a look at [past
+commits](https://github.com/bcoin-org/bcoin/commits/master) for examples of
+commit messages.
+
+Here are some of the reasons why wrapping your commit messages to 72 columns is
+a good thing.
+
+- git log doesn’t do any special special wrapping of the commit messages. With
+  the default pager of less -S, this means your paragraphs flow far off the edge
+  of the screen, making them difficult to read. On an 80 column terminal, if we
+  subtract 4 columns for the indent on the left and 4 more for symmetry on the
+  right, we’re left with 72 columns.
+- git format-patch --stdout converts a series of commits to a series of emails,
+  using the messages for the message body.  Good email netiquette dictates we
+  wrap our plain text emails such that there’s room for a few levels of nested
+  reply indicators without overflow in an 80 column terminal.
+
+<a name="CodeApproval" />
+
+### 5. Code Approval Process
+
+This section describes the code approval process that is used for code
+contributions.  This is how to get your changes into bcoin.
+
+<a name="CodeReview" />
+
+### 5.1 Code Review
+
+All code which is submitted will need to be reviewed before inclusion into the
+master branch.  This process is performed by the project maintainers and usually
+other committers who are interested in the area you are working in as well.
+
+##### Code Review Timeframe
+
+The timeframe for a code review will vary greatly depending on factors such as
+the number of other pull requests which need to be reviewed, the size and
+complexity of the contribution, how well you followed the guidelines presented
+on this page, and how easy it is for the reviewers to digest your commits.  For
+example, if you make one monolithic commit that makes sweeping changes to things
+in multiple subsystems, it will obviously take much longer to review.  You will
+also likely be asked to split the commit into several smaller, and hence more
+manageable, commits.
+
+Keeping the above in mind, most small changes will be reviewed within a few
+days, while large or far reaching changes may take weeks.  This is a good reason
+to stick with the [Share Early, Share Often](#ShareOften) development practice
+outlined above.
+
+##### What is the review looking for?
+
+The review is mainly ensuring the code follows the [Development
```

Comment:
>The review is mainly ensuring the code follows the [Development
Practices](#DevelopmentPractices) and [Code Contribution
Standards](#Standards).  However, reviewers will also check that the contribution:
>
>- Properly uses existing APIs
>- Fits into the overall architecture of the codebase
>- Does not have stability or security concerns
>- Does not introduce behavior deemed inappropriate by the community